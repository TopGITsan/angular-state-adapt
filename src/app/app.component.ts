import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';

@Component({
  selector: 'app-root',
  imports: [RouterOutlet],
  template: ` <router-outlet /> `,
  styles: `
    :host {
      @apply h-full;
    }
  `,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent {}
