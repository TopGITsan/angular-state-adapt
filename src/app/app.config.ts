import { provideHttpClient } from '@angular/common/http';
import {
  ApplicationConfig,
  isDevMode,
  provideExperimentalCheckNoChangesForDebug,
  provideExperimentalZonelessChangeDetection,
} from '@angular/core';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { provideRouter } from '@angular/router';
import { defaultStoreProvider } from '@state-adapt/angular';
import { provideAppTransloco } from '@transloco/provide-app-transloco.provider';
import { routes } from './app.routes';
export const appConfig: ApplicationConfig = {
  providers: [
    provideExperimentalZonelessChangeDetection(),
    provideRouter(routes),
    provideAnimationsAsync(),
    defaultStoreProvider,
    // only for development purposes
    isDevMode()
      ? provideExperimentalCheckNoChangesForDebug({
          interval: 10000,
          useNgZoneOnStable: false,
          exhaustive: true,
        })
      : [],
    provideHttpClient(),
    provideAppTransloco(),
  ],
};
