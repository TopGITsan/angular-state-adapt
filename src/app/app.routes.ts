import { homePath } from '@home/home.route';
import { Routes } from '@angular/router';

export const routes: Routes = [
  {
    path: '',
    loadChildren: () =>
      import('./shell-sidenav/shell-sidenav.routes').then((m) => m.routes),
  },
  { path: '', redirectTo: `/${homePath}`, pathMatch: 'full' },
  { path: '**', redirectTo: `/${homePath}`, pathMatch: 'full' },
];
