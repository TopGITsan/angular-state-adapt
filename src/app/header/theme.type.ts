// export const themes = ['os-default', 'dark', 'light'] as const;
// export type Theme = (typeof themes)[number];
export enum Theme {
  'os-default' = 'deblur',
  'dark' = 'dark_mode',
  'light' = 'light_mode',
}
export type ThemeKey = keyof typeof Theme;
export const themeValues = Object.keys(Theme).filter(
  (k): k is ThemeKey => typeof k === 'string' && k in Theme,
);
export const themeButtonIcons = Object.values(Theme).filter(
  (k): k is Theme => typeof k === 'string',
);
