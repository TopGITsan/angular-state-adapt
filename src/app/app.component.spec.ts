import { provideExperimentalZonelessChangeDetection } from '@angular/core';
import { TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { HeaderComponent } from './header/ui/header.component';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { defaultStoreProvider } from '@state-adapt/angular';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [
        provideExperimentalZonelessChangeDetection(),
        provideAnimationsAsync(),
        defaultStoreProvider,
      ],
      imports: [AppComponent, HeaderComponent],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have the 'angular-state-adapt' title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('angular-state-adapt');
  });

  it('should render title', async () => {
    const fixture = TestBed.createComponent(AppComponent);

    await fixture.whenStable();

    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled.querySelector('h1')?.textContent).toContain(
      'Hello, angular-state-adapt',
    );
  });
});
