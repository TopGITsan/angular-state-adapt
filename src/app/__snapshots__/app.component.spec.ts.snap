// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
<app-root>
  <router-outlet />
</app-root>
`;
